### ISSUE TITLE
MTG Items in unternull

### ISSUE THREAD

**Issue by bell07** [CasimirKaPazi/unternull#4]
I think we should discuss about the development at this place and not in forum. But we can switch back if you like..
This is my which-list of Items I like to see in unternull from MTG. This discussion should be a brainstorming about the best way to get them in water.. I start

1) Flora: (flowers+ gras) Can be determinated dynamically from registered_decorations with deco_type == "simple". In addition we need the Items with group "sapling". We can reduce the initial gift to default sapling only if the oder saplings are obtainable.
My proposal is to add them as very rare byproduct in "composting"-mod, or a new "mutagen-furnance" mod. Seeds could be obtained in usual way after grass is growing. dirt_with_* should be appear trough grass too.

2) Desert Stone: Maybe reading biome data for coordinates and apply the stone selection to lava cooling. So the lava provide ether stone or desert stone? The desert stone can be processed to desert sand in grinder. In case of desert stone at the time no ores are defined in MTG so maybe the player need to search an other place for stone production.

3 Ice. As "the Stone of Ice Biome" in lavacooling is to strange for me too. How to get it?
Ice can be grinded to snow.

4. Silver sand. There is no silver stone yet. that can be grinded.

Did I something missed? Any Ideas how to get it all?

**Comment by CasimirKaPazi** [comment]
I think the main problem is that I tried to support two subgames at the same time. Voxelgarden has several ways to get things you don't get in standard MTG. Adding them again in this mod would be redundant. Grass and flowers for example can already be made in VG, but silver sand on the other side does not exist.

I think the best is to split this in two mods "unternullMTG" and "unternullVG". This would also remove all the checks for different nodes.

**Comment by bell07** [comment]
Maybe my point of view is "too generic" (again). My understanding is: the Unternull is a modpack that allow to play survival in a world without anything generated by mapgen. That means the modpack contains some additional ways to get Items available (rare) that are usually provided by mapgen.

So I do not see the reason to avoid redundant ways to get items if the ways are rare or with effort. The grinder as example is still usable and may useful with default mapgen, ok, will not be used in most cases because the getting of mapgen provided items is simpler. Anyway it does not disturb the gameplay.

I am not about mods or subgames. I am about mapgen provided items. A small part: registered_ores are done in lava cooling + partially in grinder. Now I am working on registered_decorations to cover the "1. Flora". I forked your fork of the compost mod and did already some changes, expect to finish them this night. By the way, did you planned to add the modified compost mod to this modpack?

I play the "private mods collection subgame". Ok, the mtg is included as modpack. The VG-subgame I did not tried. If you see the VG does usable ways to get Items I search for, let me know. In which way the VG provides the flora?

At the end it is your decision what you need to apply to the modpack. My interest is temporary as long my childs and me plays the private "Unternull-World" in home LAN.

**Comment by bell07** [comment]
1. Flora - done in https://github.com/CasimirKaPazi/compost/pull/1

**Comment by bell07** [comment]
2. Desert and silver - done by simple recipes in https://github.com/CasimirKaPazi/unternull/pull/5

**Comment by bell07** [comment]
Still no idea for ice and snow. Ice could be grinded to snow and snow mixed with water could be Ice. But one of them we need farmable in any way. Ideas?

**Comment by CasimirKaPazi** [comment]
Maybe ice could be done by mapgen. I'll have to look into it.

Back to the idea that we split unternull into two, one for each subgame. Then you could maintain the MTG version and I the VG version. The thing is that much of unternull is already in VG (and a lot is moving from unternull to VG because it proved to be usefull), a modpack specific for the subgame would be quite different than a general one.

To sum up my idea:
- unternullMTG with your compost mod, maintained by you (and possibly others).
- unternullVG which will make mostly use of features in VG and limit itself to the mapgen and initial chest.

But I have little time this week, so don't expect anything to happen from my side.

**Comment by bell07** [comment]
I do not play MTG, it is just the base of my "Private mods collection". If you know a way to obtain Items I search for (in VG-game?) let me know. If I do not know the way I develop an own and propose it ;-) But I am not really interested to maintain (and be responsible for) anything. So you can just cherry-pick changes you like and reject changes you do not like for VG.

About Ice in mapgen, Can you explain the idea?

**Comment by bell07** [comment]
Another question. Instead of splitting anything maybe the beter idea is to start an own subgame, based on VG? Because a modpack should be generic to be usable with any subgame.

**Comment by bell07** [comment]
Started new mod to get Ice in unternull available: 
https://github.com/bell07/minetest-frozenenergy
Added crafting recipe for snow+sand=silver_sand; grinding Ice to snow in 
https://github.com/bell07/minetest-unternull/commit/cf3abdf1d80e1522d82826dd7bd308cb0353510c
For me the 1-4 list of this issue is done for now.

About the modpack and maintenance: I prefer to maintain each mod separately, so my versions of compost and frozenenergy I will maintain in this way. Of course I can take PR's to your modpack if you like it. What do you see in the unternull from this changes?

What is the origin home for crushingfurnace mod? So I can propose my changes to the origin mod too.

**Comment by bell07** [comment]
Just for reference, if you blacklisted the diamonds in the lava cooling you can use the compressor https://forum.minetest.net/viewtopic.php?f=9&t=14557 to get diamonds from coal

**Comment by bell07** [comment]
I did sorted all my changes and created push requests for all of them. Feel free to accept or reject them, or keep for future use. The most of the changes does have a "soft" dependency to other mods that can provide items (ice, saplings) not included in modpack. I referenced all my mods I use in my two worlds (Übernull und Unternull) for that reason. As already said you can add them to the modpack if you like, but it is not mandatory. Maybe it is a good idea to collect optional compatible/usefull mods in the README file as a list for reference?
